src/manifolds/Segre.jl

```diff
@@ -0,0 +1,473 @@
+
+The geometry is summarized in [JacobssonSwijsenVandervekenVannieuwenhoven:2024](@cite).
+
+# Constructor
+    Segre(valence::NTuple{V, Int}; field::AbstractNumbers=ℝ)
+
+Generate a valence `V` Segre manifold.
+"""
+struct Segre{𝔽,V} <: AbstractManifold{𝔽} end
+
+function Segre(valence::NTuple{V,Int}; field::AbstractNumbers=ℝ) where {V}
+    return Segre{field,valence}()
+end
+
+valence(::Segre{𝔽,V}) where {𝔽,V} = V
+ndims(::Segre{𝔽,V}) where {𝔽,V} = length(V)
+manifold_dimension(::Segre{𝔽,V}) where {𝔽,V} = (1 + sum(V .- 1))
+
+"""
+    check_size(M::Segre{𝔽, V}, p)
+
+Check whether `p` has the right size for `Segre` manifold `M`.
+"""
+function check_size(M::Segre{𝔽,V}, p;) where {𝔽,V}
+    p_size = only.(size.(p))
+    M_size = [1, V...]
+
+    if p_size != M_size
+        return DomainError(
+            p_size,
+            "The point $(p) can not belong to the manifold $(M), since its size $(p_size) is not equal to the manifolds representation size ($(M_size)).",
+        )
+    end
+
+    return nothing
+end
+
+"""
+    check_size(M::Segre{𝔽, V}, p, v)
+
+Check whether `p` and `v` have the right size for `Segre` manifold `M`.
+"""
+function check_size(M::Segre{𝔽,V}, p, v;) where {𝔽,V}
+    p_size = only.(size.(p))
+    v_size = only.(size.(v))
+    M_size = [1, V...]
+
+    if p_size != M_size
+        return DomainError(
+            p_size,
+            "The point $(p) can not belong to the manifold $(M), since its size $(p_size) is not equal to the manifolds representation size ($(M_size)).",
+        )
+    end
+
+    if v_size != M_size
+        return DomainError(
+            v_size,
+            "The vector $(v) can not belong to the manifold $(M), since its size $(v_size) is not equal to the manifolds representation size ($(M_size)).",
+        )
+    end
+
+    return nothing
+end
+
+"""
+    check_point(M::Segre{ℝ, V}, p; kwargs...)
+
+Check whether `p` is a valid point on `M`, i.e. `p[1]` is a singleton containing a positive number and `p[i + 1]` is a point on `Sphere(V[i])`. The tolerance can be set using the `kwargs...`.
+"""
+function check_point(M::Segre{ℝ,V}, p; kwargs...) where {V}
+    if p[1][1] <= 0.0
+        return DomainError(p[1][1], "$(p) has non-positive modulus.")
+    end
+
+    for (x, n) in zip(p[2:end], V)
+        e = check_point(Sphere(n - 1)::AbstractSphere, x; rtol=1e-10, atol=1e-10, kwargs...)
+        if !isnothing(e)
+            return e
+        end
+    end
+
+    return nothing
+end
+
+"""
+    function check_vector(M::Segre{ℝ, V}, p, v, kwargs...)
+
+Check whether `v` is a tangent vector to `p` on `M`, i.e. after `check_point(M, p)`, `v` has to be of same dimension as `p` and orthogonal to `p`. The tolerance can be set using the `kwargs...`.
+"""
+function check_vector(M::Segre{ℝ,V}, p, v, kwargs...) where {V}
+    e = check_point(M, p, kwargs...)
+    if !isnothing(e)
+        return e
+    end
```

Comment:
```suggestion
```
This should not be done here; `is_vector` has an option to enable or disable the check and does perform this. for the `check_vector` function the idea is hence that either the user did not want to check it or in other words we may assume this is correct anyways.